tests/test_preprints.py

```diff
@@ -462,13 +467,114 @@ def test_format_preprint(self):
 
         assert nodes == {}
 
+    def test_format_thesis(self):
+        res = format_preprint(self.thesis, self.thesis.provider.push_share_type)
+
+        assert set(gn['@type'] for gn in res) == {'creator', 'contributor', 'throughsubjects', 'subject', 'throughtags', 'tag', 'workidentifier', 'agentidentifier', 'person', 'thesis', 'workrelation', 'creativework'}
```

Comment:
Minor, personal preference that set tests check lengths as well unless dupes are acceptable.